golang/scripts/build.sh

```diff
@@ -10,11 +10,66 @@ root_dirpath="$(dirname "${script_dirpath}")"
 # ==================================================================================================
 #                                             Constants
 # ==================================================================================================
-KURTOSIS_DOCKERHUB_ORG="kurtosistech"
-IMAGE_NAME="kurtosis-lambda-starter-pack"
+# A sed regex that will be used to determine if the user-supplied image name matches the regex
+ALLOWED_IMAGE_NAME_CHARS='a-z0-9._/-'
 
+KURTOSIS_LAMBDA_FOLDER="kurtosis-lambda"
 
-# ==================================================================================================
-#                                             Main Logic
-# ==================================================================================================
-docker build -t "${KURTOSIS_DOCKERHUB_ORG}/${IMAGE_NAME}" --progress=plain -f "${root_dirpath}/Dockerfile" "${root_dirpath}"
+
+# =============================================================================
+#                             Pre-Arg Parsing
+# =============================================================================
+show_help_and_exit() {
```

Comment:
I think this was a good impetus, though I think this script can probably be a lot simpler - really just a hardcoded output Docker image name and a `docker build` step (and maybe the Git tag-detecting stuff)